db/migrations/1657379568_create_courses_table.up.sql

```diff
@@ -16,9 +16,11 @@ CREATE TABLE courses
     deleted_at      timestamp
 );
 
-CREATE UNIQUE INDEX courses_id_uindex
-    ON courses (id);
+COMMENT ON COLUMN courses.deleted_at IS 'Timestamp indicating when a course was softly deleted, allowing for data recovery and historical queries without permanently removing the record from the database. A NULL value means the subscription is active.';
```

Comment:
Course not subscription

Reply:
I added a unique index based on the code and a "deleted_at" field because I anticipate that we might have duplicate codes as long as one of them has been marked as deleted:

```
CREATE UNIQUE INDEX courses_code_uindex
    ON courses (code, deleted_at) NULLS NOT DISTINCT;
```

Reply:
Using soft delete means we can only update records that have not been deleted. If someone wants to modify a deleted record, they must first reactivate the record, then make the desired changes, and finally, if necessary, delete it again. This method preserves the integrity of historical data and allows for the recovery or analysis of deleted entries. However, it also requires additional logic in the application to distinguish between active and deleted records, especially during update operations. The process of reactivating, modifying, and optionally re-deleting a record might seem cumbersome but it's essential for maintaining data consistency and adhering to the soft delete pattern.

Reply:
I know what soft delete is, but we need to decide whether deleted_at will have the archived behavior or not. We can change archived data, by unarchiving for example.

Reply:
If we are aligned, I will just update our comment on this column.

Reply:
https://github.com/sumelms/microservice-course/pull/19/commits/c04e66a4cce0b4bf28f7e13db39c5d4d1c41e173

Reply:
Just remove the historical part in the description the rest is great